src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/jpa/entity/BookingEntity.kt

```diff
@@ -129,14 +130,30 @@ interface BookingRepository : JpaRepository<BookingEntity, UUID> {
   fun findByBedIdAndArrivingBeforeDate(bedId: UUID, date: LocalDate, thisEntityId: UUID?): List<BookingEntity>
 
   @Query(
-    "SELECT b FROM BookingEntity b WHERE (b.bed, b.departureDate) IN (" +
-      "  SELECT b2.bed, MAX(b2.departureDate)" +
-      "  FROM BookingEntity b2 " +
-      "  WHERE b2.departureDate <= :date " +
-      "  AND b2.bed.id IN :bedIds " +
-      "  AND SIZE(b2.cancellations) = 0 " +
-      "  GROUP BY b2.bed " +
-      ")",
+    """
```

Comment:
As we are moving away from a standard JPQL here to optimise the query - could you add a couple of comments in on the query to highlight exactly what is doing what.
I think that may improve readability for an engineering coming fresh to this code, if they need to increase the scope of this search. 

```
... SELECT b2.bed_id,                       -- Select the bed_id
        MAX(b2.departure_date) AS departure_date -- Find the latest departure date for each bed
        FROM bookings b2 ...
```

Aware we don't use comments too much in the API code, but unsure why

Reply:
Comment added. I don't prefer to add comments in the code it should be readable except it its complicated to understand

Reply:
Thanks @daveawc that is good point

Reply:
@muhammad-elabdulla I believe using Exists with the subquery is more performant than count(*) > 0

Reply:
@awhodgson256 I will replace it with left join

Reply:
@daveawc  this query is missing some rows compare with the original one